2024-04-16 13:28:41,370: root: INFO: MLTrader : Executing the initialize lifecycle method
2024-04-16 13:28:42,009: root: INFO: Current backtesting datetime 2023-12-15 08:30:00-05:00
2024-04-16 13:28:42,009: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-04-16 13:28:42,010: root: INFO: Current backtesting datetime 2023-12-15 09:30:00-05:00
2024-04-16 13:28:42,010: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-04-16 13:28:42,011: root: INFO: [34mMLTrader : Executing the on_trading_iteration lifecycle method at 2024-04-16 13:28:42[0m
2024-04-16 13:28:42,011: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-04-16 13:28:42,015: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-04-16 13:28:42,291: root: ERROR: 'NewsV2' object has no attribute '_dict_'
2024-04-16 13:28:42,294: root: ERROR: Traceback (most recent call last):
  File "/Users/nouhbouhajib/.pyenv/versions/3.8.16/lib/python3.8/site-packages/lumibot/strategies/strategy_executor.py", line 829, in run
    self._run_trading_session()
  File "/Users/nouhbouhajib/.pyenv/versions/3.8.16/lib/python3.8/site-packages/lumibot/strategies/strategy_executor.py", line 780, in _run_trading_session
    self._on_trading_iteration()
  File "/Users/nouhbouhajib/.pyenv/versions/3.8.16/lib/python3.8/site-packages/lumibot/strategies/strategy_executor.py", line 274, in func_output
    result = func_input(self, *args, **kwargs)
  File "/Users/nouhbouhajib/.pyenv/versions/3.8.16/lib/python3.8/site-packages/lumibot/strategies/strategy_executor.py", line 298, in func_output
    result = func_input(self, *args, **kwargs)
  File "/Users/nouhbouhajib/.pyenv/versions/3.8.16/lib/python3.8/site-packages/lumibot/strategies/strategy_executor.py", line 87, in func_output
    return func_input(self, *args, **kwargs)
  File "/Users/nouhbouhajib/.pyenv/versions/3.8.16/lib/python3.8/site-packages/lumibot/strategies/strategy_executor.py", line 378, in _on_trading_iteration
    on_trading_iteration()
  File "/Users/nouhbouhajib/.pyenv/versions/3.8.16/lib/python3.8/site-packages/lumibot/tools/decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
  File "/Users/nouhbouhajib/.pyenv/versions/3.8.16/lib/python3.8/site-packages/lumibot/tools/decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
  File "traidingbot.py", line 52, in on_trading_iteration
    news = self.get_the_news()
  File "traidingbot.py", line 45, in get_the_news
    news = [ev._dict_["_raw"]["headline"] for ev in news ]
  File "traidingbot.py", line 45, in <listcomp>
    news = [ev._dict_["_raw"]["headline"] for ev in news ]
  File "/Users/nouhbouhajib/.pyenv/versions/3.8.16/lib/python3.8/site-packages/alpaca_trade_api/entity.py", line 29, in __getattr__
    return super().__getattribute__(key)
AttributeError: 'NewsV2' object has no attribute '_dict_'

2024-04-16 13:28:42,294: root: INFO: MLTrader : Executing the on_bot_crash event method
2024-04-16 13:28:42,299: root: INFO: Backtesting finished
2024-04-16 13:28:42,301: root: WARNING: Cannot plot returns because the strategy returns are missing
2024-04-16 13:28:42,364: root: WARNING: Cannot create a tearsheet because the strategy returns are missing
2024-04-16 13:28:42,364: backtest_stats: INFO: Backtest took 0:00:00.995257 for a speed of 0.000
